DFE-Digital/apply-for-teacher-training
# Refactor provider user permissions forms

## Context
There was a bug on the provider user invitation wizard whereby the error summary did not link to the correct place.
There is lots of duplication between the wizard there and in the single provider user permission editing form.

## Changes proposed in this pull request
Move the edit provider user permission form to use `fields_for`
Move the wizard so it uses the same form html and ruby model
This commonises the erb and the validations

To test:
In the review app:
Invite a new provider user (Your account > Users > Invite User). Go through the flow and ensure checking different things has the desired effect.
Edit a provider user's permissions. Go to Your account> Users > click a user > click Change next to one of their permissions for a provider. Make sure that checking different boxes and submitting has the desired effect


## Guidance to review
Per commit
Are there any edge cases I've missed from the specs?
(Note that this should result in no change in behaviour apart from that the error is correctly linked to when selecting "Extra permissions" and then not checking any boxes)

## Link to Trello card
https://trello.com/c/EW62DVmA/3421-dac-change-no-errors-when-selecting-permissions

## Things to check

- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)
